tanner/tests/test_api.py

```diff
@@ -0,0 +1,224 @@
+
+    def test_return_session_info(self):
+        self.handler = Api(self.redis_client)
+
+        sessions = {
+            "sess1": {
+                "sess_uuid": "c546114f97f548f982756495f963e280"
+            },
+            "sess2": {
+                "sess_uuid": "f432785f97f548f98289054f963e972"
+            }
+        }
+        self.handler.return_snare_info = AsyncMock(return_value=sessions)
+        self.expected_content = 'sess1'
+
+        async def test():
+            self.returned_content = await self.handler.return_session_info(self.uuid, self.snare_uuid)
+
+        self.loop.run_until_complete(test())
+        self.assertEqual(self.returned_content, self.expected_content)
+
+    def test_return_sessions(self):
+        self.handler = Api(self.redis_client)
+        self.handler.return_snares = AsyncMock(return_value=["8fa6aa98-4283-4085-bfb9-a1cd3a9e56e4"])
+
+        sessions = {
+            "sess1": {
+                'user_agent': "Mozilla/5.0",
+                'peer_ip': "10.0.0.3"
+            },
+            "sess2": {
+                'attack_types': ["lfi", "xss"],
+                'possible_owners': ["crawler"],
+                'start_time': 148580,
+                'end_time': 148588,
+                'snare_uuid': [self.snare_uuid]
+            }
+        }
+
+        self.handler.return_snare_info = AsyncMock(return_value=sessions)
+        self.filters = {
+            'user_agent': "Mozilla",
+            'peer_ip': "10.0.0.1",
+            'attack_types': "xss",
+            'possible_owners': "crawler",
+            'start_time': 148575,
+            'end_time': 148590,
+            'snare_uuid': self.snare_uuid
+        }
+
+        def mock_apply_filter(filter_name, filter_value, session):
+            return True
+
+        self.handler.apply_filter = mock_apply_filter
+        self.expected_content = ["sess1", "sess2"]
+
+        async def test():
+            self.returned_content = await self.handler.return_sessions(self.filters)
+
+        self.loop.run_until_complete(test())
+        self.assertEqual(self.expected_content, self.returned_content)
+
+    def test_return_sessions_error(self):
+        self.handler = Api(self.redis_client)
+        self.handler.return_snares = AsyncMock(return_value=["8fa6aa98-4283-4085-bfb9-a1cd3a9e56e4"])
+
+        session = {
+            "sess1": {
+                "attack_types": ["rfi", "lfi"]
+            }
+        }
+        self.handler.return_snare_info = AsyncMock(return_value=session)
+
+        self.filters = {
+            "attacktypes": "lfi"
+        }
+
+        self.expected_content = 'Invalid filter : attacktypes'
+
+        async def test():
+            self.returned_content = await self.handler.return_sessions(self.filters)
+
+        self.loop.run_until_complete(test())
+        self.assertEqual(self.expected_content, self.returned_content)
+
+    def test_apply_filter_user_agent(self):
```

Comment:
Add some false results as well.

Reply:
What about false results for this method?

Reply:
On it!